File: cs/lc303.cs
public class NumArray {
    private int[] arr;
    public NumArray(int[] nums) {
        this.arr = new int[nums.Length];
        this.arr[0] = nums[0];
        for(int i=1;i<nums.Length;i++)
        {
            this.arr[i] = this.arr[i-1] + nums[i];
        }
    }
    
    public int SumRange(int left, int right) {
        if(left >0)
            return this.arr[right] - this.arr[left-1];
        else 
            return this.arr[right];
    }
}

File: python/lc303.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.arr = [0] * len(nums)
        prev= 0
        for i in range(len(nums)):
            prev += nums[i]
            self.arr[i] = prev
        

    def sumRange(self, left: int, right: int) -> int:        
        return self.arr[right]- (self.arr[left-1] if left > 0 else 0)
        
